File: bin/Debug/net5.0/v4.4.2/lgcy-on-lgcy/examples/t_api-captcha-mvc4-api_basics/razor/csharp/Controllers/ExampleController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

using Mvc40BasicCaptchaExampleRazorCSharp.Models;

using BotDetect.Web.Mvc;

namespace Mvc40BasicCaptchaExampleRazorCSharp.Controllers
{
    public class ExampleController : Controller
    {
        //
        // GET: /Example/

        public ActionResult Index()
        {
            return View();
        }


        //
        // POST: /Example/

        [HttpPost]
        [CaptchaValidationActionFilter("CaptchaCode", "ExampleCaptcha", "Incorrect CAPTCHA code!")]
        public ActionResult Index(ExampleModel model)
        {
            MvcCaptcha.ResetCaptcha("ExampleCaptcha");
            return View(model);
        }

    }
}

File: bin/Debug/net5.0/v4.4.2/lgcy-on-lgcy/examples/t_api-captcha-mvc4-api_basics/razor/vbnet/Controllers/ExampleController.vb
Imports BotDetect.Web.Mvc
Imports System.Web.Mvc

Namespace Mvc40BasicCaptchaExampleRazorVBNet
    Public Class ExampleController
        Inherits System.Web.Mvc.Controller

        '
        ' GET: /Example

        Function Index() As ActionResult
            Return View()
        End Function


        '
        ' POST: /Example/

        <HttpPost()> _
        <CaptchaValidationActionFilter("CaptchaCode", "ExampleCaptcha", "Incorrect CAPTCHA code!")> _
        Public Function Index(ByVal model As ExampleModel) As ActionResult
            MvcCaptcha.ResetCaptcha("ExampleCaptcha")
            Return View(model)
        End Function

    End Class
End Namespace